test.js

```diff
@@ -5,27 +5,202 @@ import tempy from 'tempy';
+
+	t.is(fileType(buffers[0]).ext, 'png');
+	t.is(fileType(buffers[1]).ext, 'png');
+});
+
 test('app process id', async t => {
-	const {stdout} = await execa('pgrep', ['Finder']);
-	const finderPid = parseInt(stdout.split('\n')[0], 10);
+	const finderPid = await processId('Finder');
+
 	t.is(fileType(await fileIcon.buffer(finderPid)).ext, 'png');
 });
 
+test('array of single app process id', async t => {
+	const finderPid = await processId('Finder');
+
+	const pids = [finderPid];
+	const buffers = await fileIcon.buffer(pids);
+
+	t.is(fileType(buffers[0]).ext, 'png');
+});
+
+test('array of multiple app process ids', async t => {
+	const apps = ['Finder', 'loginwindow'];
+	const pids = await Promise.all(apps.map(processId));
+	const buffers = await fileIcon.buffer(pids);
+
+	t.is(fileType(buffers[0]).ext, 'png');
+	t.is(fileType(buffers[1]).ext, 'png');
+});
+
+test('array of mixed types', async t => {
+	const loginwindowPid = await processId('loginwindow');
+
+	const apps = ['com.apple.Finder', loginwindowPid, 'Safari'];
+	const buffers = await fileIcon.buffer(apps);
+
+	t.is(fileType(buffers[0]).ext, 'png');
+	t.is(fileType(buffers[1]).ext, 'png');
+	t.is(fileType(buffers[2]).ext, 'png');
+});
+
 test('file path', async t => {
 	t.is(fileType(await fileIcon.buffer('/Applications/Safari.app')).ext, 'png');
 });
 
-test('write file', async t => {
-	const destination = tempy.file({extension: 'png'});
+test('write file single app name', async t => {
+	const destination = tmpFile();
 	await fileIcon.file('Safari', {destination});
 	const icon = fs.readFileSync(destination);
 	t.is(fileType(icon).ext, 'png');
 });
+
+test('write file array of single app name', async t => {
+	const destination = [tmpFile()];
+
+	await fileIcon.file(['Safari'], {destination});
+
+	t.is(fileType(fs.readFileSync(destination[0])).ext, 'png');
+});
+
+test('write files array of app names', async t => {
+	const destination = [tmpFile(), tmpFile()];
+
+	await fileIcon.file(['Safari', 'Finder'], {destination});
+
+	t.is(fileType(fs.readFileSync(destination[0])).ext, 'png');
+	t.is(fileType(fs.readFileSync(destination[1])).ext, 'png');
+});
+
+test('write file single app bundle id', async t => {
+	const destination = tmpFile();
+	await fileIcon.file('com.apple.Safari', {destination});
+	const icon = fs.readFileSync(destination);
+	t.is(fileType(icon).ext, 'png');
+});
+
+test('write file array of single app bundle id', async t => {
+	const destination = [tmpFile()];
+
+	await fileIcon.file(['com.apple.Safari'], {destination});
+
+	t.is(fileType(fs.readFileSync(destination[0])).ext, 'png');
+});
+
+test('write files array of app bundle ids', async t => {
+	const destination = [tmpFile(), tmpFile()];
+
+	await fileIcon.file(['com.apple.Safari', 'com.apple.Finder'], {destination});
+
+	t.is(fileType(fs.readFileSync(destination[0])).ext, 'png');
+	t.is(fileType(fs.readFileSync(destination[1])).ext, 'png');
+});
+
+test('write file single app process id', async t => {
```

Comment:
```suggestion
test('write file single app process ID', async t => {
```

id => ID, everywhere
pid => PID, everywhere